### ISSUE TITLE
Flask application compilation

### ISSUE THREAD

**Issue by shtock** [Nuitka/Nuitka#700]
Hello
I created the app.py file with

```
from flask import Flask
from sys import exit
import time

app = Flask(__name__)

@app.route("/")
def hello():    
    return "Hello World!"



if __name__ == '__main__':
    app.run(debug=True)

```

Next I compile it via

`python -m nuitka --standalone --mingw64 --follow-imports app.py`
```

Nuitka:WARNING:Unresolved '__import__' call at 'C:\Users\myuser\AppData\Local\Programs\Python\Python37\lib\site-packages\flask\cli.py:240' may require use of '--include-plugin-directory' or '--include-plugin-files'.
Nuitka:WARNING:Unresolved '__import__' call at 'C:\Users\myuser\AppData\Local\Programs\Python\Python37\lib\site-packages\flask\helpers.py:800' may require use of '--include-plugin-directory' or '--include-plugin-files'.
Nuitka:WARNING:Unresolved '__import__' call at 'C:\Users\myuser\AppData\Local\Programs\Python\Python37\lib\site-packages\flask\helpers.py:895' may require use of '--include-plugin-directory' or '--include-plugin-files'.
Nuitka:WARNING:Unresolved '__import__' call at 'C:\Users\myuser\AppData\Local\Programs\Python\Python37\lib\site-packages\jinja2\loaders.py:494' may require use of '--include-plugin-directory' or '--include-plugin-files'.
Nuitka:WARNING:Unresolved '__import__' call at 'C:\Users\myuser\AppData\Local\Programs\Python\Python37\lib\site-packages\jinja2\utils.py:142' may require use of '--include-plugin-directory' or '--include-plugin-files'.
Nuitka:WARNING:Unresolved '__import__' call at 'C:\Users\myuser\AppData\Local\Programs\Python\Python37\lib\site-packages\jinja2\utils.py:143' may require use of '--include-plugin-directory' or '--include-plugin-files'.
Nuitka:WARNING:Unresolved '__import__' call at 'C:\Users\myuser\AppData\Local\Programs\Python\Python37\lib\site-packages\werkzeug\utils.py:568' may require use of '--include-plugin-directory' or '--include-plugin-files'.
Nuitka:WARNING:Unresolved '__import__' call at 'C:\Users\myuser\AppData\Local\Programs\Python\Python37\lib\site-packages\werkzeug\utils.py:576' may require use of '--include-plugin-directory' or '--include-plugin-files'.
```

When I run app.exe I recieve 

app.exe
```
Traceback (most recent call last):
  File "c:\work\nuitka\webservice\app.dist\app.py", line 14, in <module>
  File "c:\work\nuitka\webservice\app.dist\flask\app.py", line 967, in run
  File "c:\work\nuitka\webservice\app.dist\flask\app.py", line 887, in debug
  File "c:\work\nuitka\webservice\app.dist\flask\helpers.py", line 957, in __get__
  File "c:\work\nuitka\webservice\app.dist\flask\app.py", line 689, in jinja_env
  File "c:\work\nuitka\webservice\app.dist\flask\app.py", line 781, in create_jinja_environment
  File "c:\work\nuitka\webservice\app.dist\flask\templating.py", line 45, in __init__
  File "c:\work\nuitka\webservice\app.dist\jinja2\environment.py", line 359, in __init__
  File "c:\work\nuitka\webservice\app.dist\jinja2\environment.py", line 108, in load_extensions
  File "c:\work\nuitka\webservice\app.dist\jinja2\utils.py", line 143, in import_string
ModuleNotFoundError: No module named 'jinja2.ext'

```
My environment:
```

python -m nuitka --version
0.6.7
Python: 3.7.0 (v3.7.0:1bf9cc5093, Jun 27 2018, 04:59:51) [MSC v.1914 64 bit (AMD64)]
Executable: C:\Users\myuser\AppData\Local\Programs\Python\Python37\python.exe
OS: Windows
Arch: x86_64

```

Do you have any ideas how to fix it?

**Comment by shtock** [comment]
I had managed to make it runable via

`python -m nuitka --standalone --mingw64 --include-plugin-directory=C:\Users\myuser\AppData\Local\Programs\Python\Python37\lib\site-packages\jinja2 --follow-imports app.py `

but it created bunches of .pyd files

`dir
`
```
06.05.2020  19:53    <DIR>          .
06.05.2020  19:53    <DIR>          ..
06.05.2020  19:49        21 356 544 app.exe
06.05.2020  19:49           665 104 comctl32.dll
06.05.2020  19:49         2 483 360 libcrypto-1_1-x64.dll
06.05.2020  19:49           529 568 libssl-1_1-x64.dll
06.05.2020  19:47           197 272 pyexpat.pyd
06.05.2020  19:49         3 844 760 python37.dll
06.05.2020  19:47            26 776 select.pyd
06.05.2020  19:49         1 217 688 sqlite3.dll
06.05.2020  19:49         1 731 744 tcl86t.dll
06.05.2020  19:49         1 480 352 tk86t.dll
06.05.2020  19:47         1 073 304 unicodedata.pyd
06.05.2020  19:49            89 752 vcruntime140.dll
06.05.2020  19:47            72 856 _asyncio.pyd
06.05.2020  19:47            89 240 _bz2.pyd
06.05.2020  19:47            22 680 _contextvars.pyd
06.05.2020  19:47           132 760 _ctypes.pyd
06.05.2020  19:47           273 048 _decimal.pyd
06.05.2020  19:47           171 160 _elementtree.pyd
06.05.2020  19:47            39 064 _hashlib.pyd
06.05.2020  19:47           257 176 _lzma.pyd
06.05.2020  19:47            39 064 _msi.pyd
06.05.2020  19:47            29 336 _multiprocessing.pyd
06.05.2020  19:47            44 184 _overlapped.pyd
06.05.2020  19:47            27 800 _queue.pyd
06.05.2020  19:47            75 928 _socket.pyd
06.05.2020  19:47            85 144 _sqlite3.pyd
06.05.2020  19:47           123 032 _ssl.pyd
06.05.2020  19:47            69 784 _tkinter.pyd
```
I would expect just an one app.exe file. Is it whether expected or not? Thanks in advance.

**Comment by kayhayen** [comment]
Your expectation of single file is not met unless #575 is solved.

The inclusion of the directory is relatively wrong, because there are ways to include specific modules, but that would require to read help output or documentation. I understand, that is too much to ask, no criticism, just a statement of fact. I myself am not a big documentation user.

But I guess, Nuitka needs to add a check for inclusion of directories that are in the import path. People shoot themselves in the feet with this a lot, although in your case it is not that way. The usage of `--include-package=jinja2` would achieve the same, or more likely, only `jinja2.ext` would make more sense. Or maybe only the extension used there.  An error should point to that option instead. This again is not so much about you, but me repeatedly seeing people using this option to solve the problem, probably because they found it first.

With the minimal reproducer however, I will be able to fix the issue in Nuitka itself too. I wonder what triggers `jinja2.ext` usage, if it's Jinja2 already or something Flask does, but this kind of hidden dependency, once discovered, is easy to add usually.

**Comment by kayhayen** [comment]
This is fixed on factory, http://nuitka.net/doc/factory.html

**Comment by kayhayen** [comment]
This is released as part of 0.6.8.1 hotfix.
